### DISCUSSION TITLE
Breaking changes in minor or patch versions of `clang-format`

### DISCUSSION THREAD

**Discussion by jidicula** [jidicula/clang-format-action#192] [Polls]
I'm not sure how often this scenario happens:

> It's possible that a formatting check workflow using this action and a pinned version of `clang-format` could break in a subsequent run if the underlying `clang-format` Ubuntu package has introduced a breaking minor or patch version change.

I want to gather some data and opinions on this so I can understand this problem better.

See also https://github.com/jidicula/clang-format-action/issues/168

👇 Vote below and leave a comment with more details if you have more to share!

**Comment by ptsouchlos** [comment]
`clang-format` `18.1.3` -> `18.1.5+` resulted in breaking changes. This is a bit of a headache because it causes out CI checks to fail when it is really a false positive.

**Reply by ptsouchlos** [reply]
Relevant bug in the llvm project https://github.com/llvm/llvm-project/issues/93034

**Reply by ptsouchlos** [reply]
Also changes between `18.1.7` and `18.1.8`

**Comment by MB-Conti** [comment]
Hi, thanks for the poll.
I ran into this issue today with clang-format version 18.

A month ago, the workflow ran fine — today it failed. The root cause was a change in the Ubuntu version used in the Docker image ([Commit 2c659fe](https://github.com/jidicula/clang-format-action/commit/2c659fea724e981e94741c0e7db663982cd7d3f4)), from Noble to Questing.

The Noble-based image installs clang-format-18 as version 18.1.3, whereas the Questing-based image installs 18.1.8 ([clang-format packages](https://www.ubuntuupdates.org/package_metas?q=clang-format-18).)

I’m adding this comment as a reference in case it’s helpful to others running into similar version drift issues.
